### ISSUE TITLE
Very slow upscaling

### ISSUE THREAD

**Issue by Necrolion99** [k4yt3x/video2x#420]
## Component Versions

Please at least fill in the release version and GUI or CLI version.

- **Video2X release version**: 4.7.0
- **Upscaler version**: 4.4.0
- **GUI version**: 2.8.0
- **CLI version**: The latest one (I cannot find this info)

## Symptom
Hello guys! This is the first time I am on this site. I downloaded this app last night to upscale a 2 minute, 11MB 480p video (2.5 ratio) with the driver being waifu2x-ncnn-vulkan. At first it took me 3 hrs for the first part, then it quickly devolved into more than 9 hrs. Is this supposed to be this long? My laptop is an Asus TUF gaming, 8GB ram, AMD Ryzen 5 3550H. It wasn't the best laptop, but I don't think it can't handle upscaling such a short video. Other similar apps like Waifu2x GUI runs just fine on my laptop (4 min for an image upscaling), so I though it was weird that running video2x would be this slow. My technical knowledge is very limited, so hope I can get an easy-to-understand answer (ELI5). Thanks a lot! 
P/s: I am typing this on my phone at 6 in the morning, so hope you guys can excuse some grammar mistakes. (Pls don't kill me for the bad screenshot). 

 

.

## Error Log or Screenshots

Please upload or paste the error log here. You may also include screenshots.
![CamScanner 12-01-2020 07 04_1](https://user-images.githubusercontent.com/75283775/100680781-ee43b580-33a4-11eb-97f5-76577c2ade0c.jpg)

**It is highly recommended to include your error log.**

**Comment by Necrolion99** [comment]
Also, at the time I fi ished writing this, the time has jumped to 16 hrs. Please help!

**Comment by k4yt3x** [comment]
> My laptop is an Asus TUF gaming, 8GB ram, AMD Ryzen 5 3550H

What GPU do you have?

> 4 min for an image upscaling

A two-minute video may contain tens of thousands of frames. It will take significantly longer to upscale videos than images.

What settings did you change?

**Comment by k4yt3x** [comment]
Also I believe waifu2x-ncnn-vulkan can only do 2x scaling, which means a 2.5x upscale will actually run 2x upscaling twice.

**Comment by Necrolion99** [comment]
My GPU is Radeon RC 560X and AMD Radeon(TM) Vega 8 Graphic. I didn't change anything beside the scale ratio from 2 to 2.5. Maybe that's the problem. I will wait for this video to be done, then try again.

**Comment by Necrolion99** [comment]
So I cancelled the previous video because apparently it is just getting longer. I try using anime4k now and apparently the processes tab automatically moves to 16 when I throw the video in. Currently it only takes about 10 min. Maybe that's the problem. How do I closed thia thread as "solved"?

**Comment by k4yt3x** [comment]
Anime4K is not based on machine learning. Rather, it's closer to a traditional filter (?)

You will get much worse quality using A4K than waifu2x or other ML-based drivers. I do not suggest you to use it unless you want something done quick. If you care about quality, don't use it.

Also, A4K benefits tremendously from multi-threading. Using 16 threads is recommended for A4K. 1 thread is recommended for everything else. This is why the number of threads changes to 16 automatically when you choose A4K.

---

Vega 8 isn't competent at all. Don't use it.

RX 560X is not a compelling card for these things either. ML requires an insane amount of VRAM and computing power. I think the number of hours you're getting is quite normal.

---

Considering you're asking for a 2.5x output, v2x will first have to calculate a 4x output with two 2x iterations, which is to say:

- 480P -> 960P
- 960P -> 1920P

Larger the input file, the more time it will take. Watch for the number of passes after the file name for an idea which iteration it's in right now. If it's taking you 4 minutes to upscale a single image from 480P to 960P, then it would take you 4 * 3112 / 60 == 207 hours to upscale the entire video just two times. 4 minutes/image is way too slow. Check [this page](https://github.com/K4YT3X/video2x/wiki/Performance-Benchmarks) for some benchmarks done with the standard testing clip. A 2080Ti can upscale 240P frames to 1080P frames at around 0.9 frames/second. Notice the unit here.

There's also a possibility that waifu2x isn't really using your RX560X. I recommend you to start the upscale and observe your GPU usage in the task manager. If everything is configured properly, you should see RX560X's usage explode.

**Comment by Necrolion99** [comment]
> Anime4K is not based on machine learning. Rather, it's closer to a traditional filter (?)
> 
> You will get much worse quality using A4K than waifu2x or other ML-based drivers. I do not suggest you to use it unless you want something done quick. If you care about quality, don't use it.
> 
> Also, A4K benefits tremendously from multi-threading. Using 16 threads is recommended for A4K. 1 thread is recommended for everything else. This is why the number of threads changes to 16 automatically when you choose A4K.
> 
> ---
> 
> Vega 8 isn't competent at all. Don't use it.
> 
> RX 560X is not a compelling card for these things either. ML requires an insane amount of VRAM and computing power. I think the number of hours you're getting is quite normal.
> 
> ---
> 
> Considering you're asking for a 2.5x output, v2x will first have to calculate a 4x output with two 2x iterations, which is to say:
> 
> - 480P -> 960P
> - 960P -> 1920P
> 
> Larger the input file, the more time it will take. Watch for the number of passes after the file name for an idea which iteration it's in right now. If it's taking you 4 minutes to upscale a single image from 480P to 960P, then it would take you 4 * 3112 / 60 == 207 hours to upscale the entire video just two times. 4 minutes/image is way too slow. Check [this page](https://github.com/K4YT3X/video2x/wiki/Performance-Benchmarks) for some benchmarks done with the standard testing clip. A 2080Ti can upscale 240P frames to 1080P frames at around 0.9 frames/second. Notice the unit here.
> 
> There's also a possibility that waifu2x isn't really using your RX560X. I recommend you to start the upscale and observe your GPU usage in the task manager. If everything is configured properly, you should see RX560X's usage explode.

Well, I doubt I can afford high-tier laptops, nor am I competent enough to replace my laptop's graphic card. I did check the performance tab and apparently it is using 2% of the Vega 8, and the RX 560X is at 0%. CPU ia at 90%+, memory is 50%, disk is 50~70% (I am running a few other apps atm). Is this bad? How do I correctly configure this?

**Comment by Necrolion99** [comment]
So I do as the internet told me. I go to display and change graphic setting to high performance (using RX 560X) for a bunch of games and apps, including video2x_gui.exe. Nothing changes. Still takes 3hrs+ for the same video. Task Manager didn't even show which GPU is being used in the Processes tab, and the GPU 0 (RX 560X) is still at 0%. Am I doing this wrong? Any help is appreciated.

**Comment by tomini** [comment]
Hello,
I have the same issue when doing 240p to 1440p on my 1050Ti. I don't see any usage on GPU in Task Manager and CPU isn't used much, too.
Any tips?
Thanks
